x-pack/plugins/task_manager/server/lib/create_managed_configuration.ts

```diff
@@ -112,52 +118,66 @@ function createCapacityScan(config: TaskManagerConfig, logger: Logger, startingC
-      newPollInterval = Math.min(
-        Math.ceil(previousPollInterval * POLL_INTERVAL_INCREASE_PERCENTAGE),
-        Math.ceil(Math.max(PREFERRED_MAX_POLL_INTERVAL, startingPollInterval))
-      );
-      if (!Number.isSafeInteger(newPollInterval) || newPollInterval < 0) {
-        logger.error(
-          `Poll interval configuration had an issue calculating the new poll interval: Math.min(Math.ceil(${previousPollInterval} * ${POLL_INTERVAL_INCREASE_PERCENTAGE}), Math.max(${PREFERRED_MAX_POLL_INTERVAL}, ${startingPollInterval})) = ${newPollInterval}, will keep the poll interval unchanged (${previousPollInterval})`
-        );
-        newPollInterval = previousPollInterval;
-      }
-    } else {
-      // Decrease poll interval by POLL_INTERVAL_DECREASE_PERCENTAGE and use Math.floor to
-      // make sure the number is different than previous while not being a decimal value.
-      newPollInterval = Math.max(
-        startingPollInterval,
-        Math.floor(previousPollInterval * POLL_INTERVAL_DECREASE_PERCENTAGE)
-      );
-      if (!Number.isSafeInteger(newPollInterval) || newPollInterval < 0) {
-        logger.error(
-          `Poll interval configuration had an issue calculating the new poll interval: Math.max(${startingPollInterval}, Math.floor(${previousPollInterval} * ${POLL_INTERVAL_DECREASE_PERCENTAGE})) = ${newPollInterval}, will keep the poll interval unchanged (${previousPollInterval})`
-        );
-        newPollInterval = previousPollInterval;
+  return scan(
+    (previousPollInterval: number, { count: errorCount, isBlockException }: ErrorScanResult) => {
+      let newPollInterval: number;
+      if (isBlockException) {
+        newPollInterval = INTERVAL_AFTER_BLOCK_EXCEPTION;
+      } else {
+        if (errorCount > 0) {
+          // Increase poll interval by POLL_INTERVAL_INCREASE_PERCENTAGE and use Math.ceil to
+          // make sure the number is different than previous while not being a decimal value.
+          // Also ensure we don't go over PREFERRED_MAX_POLL_INTERVAL or startingPollInterval,
+          // whichever is greater.
+          newPollInterval = Math.min(
+            Math.ceil(previousPollInterval * POLL_INTERVAL_INCREASE_PERCENTAGE),
+            Math.ceil(Math.max(PREFERRED_MAX_POLL_INTERVAL, startingPollInterval))
+          );
+          if (!Number.isSafeInteger(newPollInterval) || newPollInterval < 0) {
+            logger.error(
+              `Poll interval configuration had an issue calculating the new poll interval: Math.min(Math.ceil(${previousPollInterval} * ${POLL_INTERVAL_INCREASE_PERCENTAGE}), Math.max(${PREFERRED_MAX_POLL_INTERVAL}, ${startingPollInterval})) = ${newPollInterval}, will keep the poll interval unchanged (${previousPollInterval})`
+            );
+            newPollInterval = previousPollInterval;
+          }
+        } else {
+          if (previousPollInterval === INTERVAL_AFTER_BLOCK_EXCEPTION) {
+            previousPollInterval = startingPollInterval;
```

Comment:
Trying to understand the logic here:

If there is no cluster block exception and the error count is 0 and the previous poll interval was the cluster block exception value, then we set the previous poll interval back to the start value? Should this be `newPollInterval = startingPollInterval` instead?

Reply:
I just didn't wanted to mess up the newPollInterval calculation 2 lines below.
It takes `startingPollInterval` and `previousPollInterval` as input and returns the newPollInterval.
But as it looks confusing, i changed it as you said.